src/main/java/com/rebuy/consul/ConsulService.java

```diff
@@ -76,6 +86,11 @@ private void verifyServiceFound(String serviceName, List<CatalogService> service
         return client.getCatalogService(serviceName, QueryParams.DEFAULT).getValue();
     }
 
+    private List<CatalogService> fetchServices(String serviceName, String tag)
+    {
+        return client.getCatalogService(serviceName, tag, QueryParams.DEFAULT).getValue();
```

Comment:
I don't think adding an extra function for one line of code increases readability.